nixos/tests/xss-lock.nix

```diff
@@ -0,0 +1,26 @@
+import ./make-test.nix ({ pkgs, lib, ... }:
+
+with lib;
+
+{
+  name = "xss-lock";
+  meta.maintainers = with pkgs.stdenv.lib.maintainers; [ ma27 ];
+
+  machine = {
+    imports = [ ./common/x11.nix ./common/user-account.nix ];
+    programs.xss-lock.enable = true;
+    programs.xss-lock.lockerCommand = "${pkgs.xlockmore}/bin/xlock";
+    services.xserver.displayManager.auto.user = "alice";
+  };
+
+  testScript = ''
+    $machine->start;
+    $machine->waitForX;
+    $machine->waitForUnit("xss-lock.service", "alice");
+
+    $machine->fail("pgrep xlock");
+    $machine->systemctl("suspend");
+    $machine->sendKeys("a"); # wakeup
```

Comment:
Suspending the VM is quite fragile and non-deterministic. This test only works if the VM does _not_ go fully into suspend yet by the time the keypress event arrives. Try `sleep 5` in the perl script (not the VM) after the `systemctl("suspend")` :  once the VM is really suspended, it doesn't come back, and the test is stuck.

As an alternative, try to activate xsslock using DPMS: `$machine->succeed("xset dpms force standby");` instead of suspend and sendKeys. This will also activate the screen locker and should give you a reliable test.